### PR TITLE
[E-algo < T581] Implement algo.cover

### PR DESCRIPTION
**Pull request by mpintaric55334**
### Description

Please briefly explain the changes you made here.

### Pull request type

- [ ] Bugfix
- [ ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

**Review by antoniofilipovic — APPROVED** [review]
Looks good to me

**Comment by antoniofilipovic** [comment]
@mpintaric55334 don't forget to update description and ping Vlasta

File: cpp/algo_module/algorithm/algo.cpp [hunk 1]
```diff
@@ -1 +1,28 @@
...
+  try {
+    auto list_nodes = arguments[0].ValueList();
+    std::unordered_set<mgp::Node> nodes;
+    for (const auto elem : list_nodes) {
+      auto node = elem.ValueNode();
+      nodes.insert(node);
+    }
+
+    for (auto node : nodes) {
+      for (auto rel : node.OutRelationships()) {
```
Comment (imilinovic) [inline cpp/algo_module/algorithm/algo.cpp:16]: You can use const auto &node and const auto rel

**Review by imilinovic — CHANGES_REQUESTED** [review]
One small change

File: cpp/algo_module/algorithm/algo.cpp [hunk 2]
```diff
@@ -1 +1,28 @@
 #include "algo.hpp"
+
+void Algo::Cover(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard{memory};
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    auto list_nodes = arguments[0].ValueList();
+    std::unordered_set<mgp::Node> nodes;
+    for (const auto elem : list_nodes) {
```
Comment (imilinovic) [inline cpp/algo_module/algorithm/algo.cpp:10]: You can use const auto &elem

**Review by imilinovic — APPROVED** [review]
Looks good to go!

**Pull request merged**